tests/helpers/zos_blockinfile_helper.py

```diff
@@ -117,8 +145,8 @@ def DsGeneral(test_name, ansible_zos_module, test_env, test_info, expected):
         results = hosts.all.shell(cmd=cmdStr)
         for result in results.contacted.values():
             pprint(result)
-            assert result.get("stdout") == expected
-            # assert result.get("stdout").replace('\n', '').replace(' ', '') == expected.replace('\n', '').replace(' ', '')
+            #assert result.get("stdout") == expected
+            assert result.get("stdout").replace('\n', '').replace(' ', '') == expected.replace('\n', '').replace(' ', '')
```

Comment:
If you replace `\n`'s for stdout and for expected and then replace ' ''s with '' for expected and stdout you have effectively balanced both sides , I have to wonder why this was needed over just using the original `result.get("stdout") == expected` ? Is there something with the `\n` and ' '  breaking our pytest cases they need to be stripped out?

Reply:
I'm not too sure, but for the datasets of type 'PDSE' sometimes mark that are not equal after a few attempts of tests

Reply:
It was a problem about the file stay there the solution it was with the deletion of file for every [test](https://github.com/ansible-collections/ibm_zos_core/blob/edd2a44eb3c80ccb22316c6c81a3b8cc983574e7/tests/helpers/zos_lineinfile_helper.py#L276)

Reply:
Thanks, so the issue was resolved when you cleaned up the test case, likely what i commented [here](https://github.com/ansible-collections/ibm_zos_core/pull/731#discussion_r1175753191)

Ill paste my comment:
```


Understood, I am not sure without some hands on investigation but I think the issue is the source file encoding is not what is iconv is being told is correct.

Eg, if for a file on your controller is encoding in UTF8 (for example) and you tell iconv -f ISO-8859-1 -t IBM-1047, its going to end up like garbage because it mapped a UTF8 file as ISO-8859-1 to IBM-1047, this is a guess but i wonder if what you had seen was just a source issue i describe.

```